src/tokenizer/tokenizer.py

```diff
@@ -1070,25 +1092,18 @@ def lookup(abbrev):
 
             clock = False
 
-            # Check for $[number]
-            if (
-                token.kind == TOK.PUNCTUATION
-                and token.txt == "$"
-                and next_token.kind == TOK.NUMBER
-            ):
-
-                token = TOK.Amount(token.txt + next_token.txt, "USD", next_token.val[0])
-                next_token = next(token_stream)
-
-            # Check for €[number]
-            if (
-                token.kind == TOK.PUNCTUATION
-                and token.txt == "€"
-                and next_token.kind == TOK.NUMBER
-            ):
-
-                token = TOK.Amount(token.txt + next_token.txt, "EUR", next_token.val[0])
-                next_token = next(token_stream)
+            # Check for currency symbol followed by number, e.g. $10
+            for symbol, currabbr in CURRENCY_SYMBOLS.items():
```

Comment:
Probably better performance-wise to drop the `for` loop and do a direct check for membership in the currency symbol `dict` (`if token.txt in CURRENCY_SYMBOLS` or something like that), then process.